src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/DumpTests.cs

```diff
@@ -0,0 +1,128 @@
+            await toolRunner.StartAsync(TestTimeouts.StartProcess);
+
+            using HttpClient httpClient = await toolRunner.CreateHttpClientDefaultAddressAsync(_httpClientFactory, TestTimeouts.HttpApi);
+            ApiClient apiClient = new(_outputHelper, httpClient);
+            
+            AppRunner appRunner = new(_outputHelper);
+            appRunner.ConnectionMode = appConnectionMode;
+            appRunner.DiagnosticPortPath = diagnosticPortPath;
+            appRunner.ScenarioName = TestAppScenarios.AsyncWait.Name;
+
+            // MachO not supported, only ELF: https://github.com/dotnet/runtime/blob/main/docs/design/coreclr/botr/xplat-minidump-generation.md#os-x
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                appRunner.Environment.Add("COMPlus_DbgEnableElfDumpOnMacOS", "1");
+            }
+
+            try
+            {
+                await appRunner.ExecuteAsync(async () =>
+                {
+                    ProcessInfo processInfo = await apiClient.GetProcessAsync(appRunner.ProcessId, TestTimeouts.HttpApi);
+                    Assert.NotNull(processInfo);
+
+                    using ApiStreamHolder holder = await apiClient.CaptureDumpAsync(appRunner.ProcessId, type, CancellationToken.None);
+                    Assert.NotNull(holder);
+
+                    const int bufferLength = 10240; // 10k buffer
+                    long total = await ArrayPool<byte>.Shared.RentAndReturnAsync(bufferLength, async buffer =>
+                    {
+                        using CancellationTokenSource cancellation = new(TestTimeouts.HttpApi);
+
+                        int read;
+                        long total = 0;
+                        while (0 != (read = await holder.Stream.ReadAsync(buffer, 0, buffer.Length, cancellation.Token)))
+                        {
+                            total += read;
+                        }
+                        return total;
+                    });
+
+                    _outputHelper.WriteLine("Dump Size: {0} bytes", total);
+                    // Dumps should have at least 10k of data (this is an arbitrary threshold).
+                    // CONSIDER: Is there a better lightweight way to check if the dump is viable?
+                    // e.g. install dotnet-dump and do a brief analysis; or check the header on the
+                    // file content.
+                    Assert.True(total > 10_000);
+
+                    await appRunner.SendCommandAsync(TestAppScenarios.AsyncWait.Commands.Continue, TestTimeouts.SendCommand);
+                });
+            }
+            catch (ApiStatusCodeException ex) when (IsSegFaultOnOSX(appRunner, ex.StatusCode))
+            {
+                // Don't fail the test if the test app segfaults due to calling the dump commmand.
+                _outputHelper.WriteLine("WARNING: Test app segfaulted while producing dump type '{0}'.", type);
```

Comment:
@josalem, I've been seeing on occasion that asking the .NET 5 runtime to produce to a dump on MacOS will segfault the runtime occasionally. Is this a known issue?

Reply:
I'm not aware of one.  @mikem8361 are you aware of anything?

Reply:
Not something we've seen.

Reply:
I'm trying to think if there's a way that I can collect the core dump of the process as part of the test; I would remove this guard to allow the test to fail so we can get that data.